### PR TITLE
Use call type on the path

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Now instead of using the call type, we add the type to the path at redux level.

I also included here the AppCall/AppCallRequest needed changes.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33178

#### Related Pull Requests
Webapp: https://github.com/mattermost/mattermost-webapp/pull/7679
Mobile: https://github.com/mattermost/mattermost-mobile/pull/5221
Proxy: https://github.com/mattermost/mattermost-plugin-apps/pull/99

### CODE REVIEW

File: src/client/client4.ts [hunk 1]
```diff
@@ -3300,11 +3300,13 @@ export default class Client4 {
     // This function belongs to the Apps Framework feature.
     // Apps Framework feature is experimental, and this function is susceptible
     // to breaking changes without pushing the major version of this package.
-    executeAppCall = async (call: AppCall) => {
-        call.context.user_agent = 'webapp';
+    executeAppCall = async (call: AppCallRequest, type: AppCallType) => {
+        const callCopy = JSON.parse(JSON.stringify(call)) as AppCallRequest;
+        callCopy.path = `${callCopy.path}/${type}`;
```
Comment (catalintomai) [inline src/client/client4.ts:3305]: should we check for `! callCopy` ?

**Review by catalintomai — APPROVED** [review]

File: src/client/client4.ts (hunk 1 above)
Comment (larkox) [reply src/client/client4.ts:3305]: I do not foresee any case where callCopy may return undefined. If call is indeed an AppCallRequest, and it is not undefined, callCopy should be always defined. Any other case I may be missing?

**Review by levb — APPROVED** [review]

File: src/client/client4.ts (hunk 1 above)
Comment (catalintomai) [reply src/client/client4.ts:3305]: Not really, just defensive coding, every time we do an `as X` we should ask the q imo.

File: src/client/client4.ts [hunk 2]
```diff
@@ -3300,11 +3300,13 @@ export default class Client4 {
     // This function belongs to the Apps Framework feature.
     // Apps Framework feature is experimental, and this function is susceptible
     // to breaking changes without pushing the major version of this package.
-    executeAppCall = async (call: AppCall) => {
-        call.context.user_agent = 'webapp';
+    executeAppCall = async (call: AppCallRequest, type: AppCallType) => {
+        const callCopy = JSON.parse(JSON.stringify(call)) as AppCallRequest;
```
Comment (mickmister) [inline src/client/client4.ts:3304]:
```suggestion
        const callCopy = {...call};
```

**Review by mickmister — CHANGES_REQUESTED** [review]

File: src/client/client4.ts [hunk 3]
```diff
@@ -3300,11 +3300,13 @@ export default class Client4 {
     // This function belongs to the Apps Framework feature.
     // Apps Framework feature is experimental, and this function is susceptible
     // to breaking changes without pushing the major version of this package.
-    executeAppCall = async (call: AppCall) => {
-        call.context.user_agent = 'webapp';
+    executeAppCall = async (call: AppCallRequest, type: AppCallType) => {
+        const callCopy = {...call};
+        callCopy.path = `${callCopy.path}/${type}`;
+        callCopy.context.user_agent = 'webapp';
```
Comment (mickmister) [inline src/client/client4.ts:3306]: 0/5 seems better to me

```suggestion
        const callCopy: AppCallRequest = {
            ...call,
            path: `${call.path}/${type}`,
            context: {
                ...call.context,
                user_agent: 'webapp',
            },
        };
```

**Review by mickmister — APPROVED** [review]

**Pull request merged**